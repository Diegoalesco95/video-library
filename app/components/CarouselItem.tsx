'use client';

import React from 'react';
import Link from 'next/link';
import Image from 'next/image';

import { MdPlayArrow, MdPlaylistAdd, MdPlaylistRemove } from 'react-icons/md';

import styles from 'app/styles/components/carousel-item.module.scss';

// import { setFavoriteUserMovie, deleteFavoriteUserMovie, getUserMovies } from '../actions';

const CarouselItem = (props) => {
	const {
		_id,
		cover,
		title,
		year,
		contentRating,
		duration,
		isList,
		userList,
		myList,
		setFavoriteUserMovie,
		deleteFavoriteUserMovie,
		getUserMovies,
	} = props;

	const handleSetFavorite = () => {
		const exist = myList.find((item) => item._id === _id);
		if (!exist) {
			setFavoriteUserMovie({
				_id,
				cover,
				title,
				year,
				contentRating,
				duration,
			});
			getUserMovies({ _id });
		} else {
		}
	};

	const handleDeleteFavorite = () => {
		const userMovieId = userList.find((userList) => userList.movieId === _id);
		deleteFavoriteUserMovie(userMovieId);
	};

	const handlePlay = (_id) => {
		// getVideoSource(_id);
	};

	return (
		<div className={styles['carousel-item']}>
			<Image className={styles['carousel-item__img']} width={200} height={250} src={cover} alt={title} />
			<div className={styles['carousel-item__details']}>
				<div className={styles['carousel-item__detail--icon']}>
					<Link href={`/player/${_id}`}>
						<MdPlayArrow className={styles.play} title='Play video' onClick={() => handlePlay(_id)} />
					</Link>
					{isList ? (
						<MdPlaylistRemove className={styles.remove} title='Delete from list' onClick={handleDeleteFavorite} />
					) : (
						<MdPlaylistAdd className={styles.add} title='Add to list' onClick={handleSetFavorite} />
					)}
				</div>
				<h2 className={styles['carousel-item__details--title']}>{title}</h2>
				<p className={styles['carousel-item__details--subtitle']}>{`${year} ${contentRating} ${duration}`}</p>
			</div>
		</div>
	);
};

export default CarouselItem;
