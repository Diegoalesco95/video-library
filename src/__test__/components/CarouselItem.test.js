import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import CarouselItem from '../../components/CarouselItem';

describe('Carousel Item component', () => {
  test('Match Snapshot', () => {
    const carousel = create(
      <ProviderMock>
        <CarouselItem />
      </ProviderMock>,
    );
    expect(carousel.toJSON()).toMatchSnapshot();
  });
});
