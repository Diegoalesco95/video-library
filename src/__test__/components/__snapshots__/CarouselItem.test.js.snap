// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Item component Match Snapshot 1`] = `
<div
  className="carousel-item"
>
  <img
    className="carousel-item__img"
  />
  <div
    className="carousel-item__details"
  >
    <div
      className="carousel-item__detail--icon"
    >
      <a
        href="/player/undefined"
        onClick={[Function]}
      >
        <img
          alt="Play"
          src="test-file.stub"
        />
      </a>
      <img
        alt="Plus"
        onClick={[Function]}
        src="test-file.stub"
      />
    </div>
    <h2
      className="carousel-item__details--title"
    />
    <p
      className="carousel-item__details--subtitle"
    >
      undefined undefined undefined
    </p>
  </div>
</div>
`;
